app/parse.py

```diff
@@ -15,9 +32,69 @@ class Product:
     num_of_reviews: int
 
 
-def get_all_products() -> None:
-    pass
+PRODUCT_FIELDS = [field.name for field in fields(Product)]
+
+
+def parse_single_product(product: WebElement) -> Product:
+    return Product(
+        title=product.find_element(By.CLASS_NAME, "title").get_property("title"),
+        description=product.find_element(By.CLASS_NAME, "description").text,
+        price=float(product.find_element(By.CLASS_NAME, "price").text.replace("$", "")),
+        rating=len(product.find_elements(By.CLASS_NAME, "ws-icon-star")),
+        num_of_reviews=int(product.find_element(By.CLASS_NAME, "review-count").text.split()[0]),
+    )
+
+
+def write_products_to_csv(output_csv_path: str, products: list[Product]) -> None:
+    with open(f"{output_csv_path}.csv", "w", newline="", encoding="utf-8") as file:
+        writer = csv.writer(file)
+        writer.writerow(PRODUCT_FIELDS)
+        writer.writerows([astuple(product) for product in products])
+
+
+def get_page_all_products() -> None:
+    with SeleniumDriverManager.get_driver() as driver:
+
+        for product, url in tqdm(URLS.items(), desc="Processing", unit="item"):
+
+            sleep(0.1)
+            print(f"Processing: {product}")
+
+            driver.get(url)
+
+            cookies_button = driver.find_elements(By.CLASS_NAME, "acceptCookies")
+
+            if cookies_button:
+                cookies_button[0].click()
+
+            more_button = driver.find_elements(
+                By.CLASS_NAME,
+                "ecomerce-items-scroll-more"
+            )
+
+            if more_button:
+                more_button = more_button[0]
+                while more_button.is_displayed():
```

Comment:
I think better to create ProductParser for example
And move all this functions to class